drivers/video/fb.c

```diff
@@ -682,6 +675,29 @@ static int fb_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
   return ret;
 }
 
+static int fb_mmap(FAR struct file *filep, FAR struct mm_map_entry_s *map)
+{
+  FAR struct inode *inode;
+  FAR struct fb_chardev_s *fb;
+  int ret = -EINVAL;
+
+  /* Get the framebuffer instance */
+
+  DEBUGASSERT(filep != NULL && filep->f_inode != NULL);
+  inode = filep->f_inode;
+  fb    = (FAR struct fb_chardev_s *)inode->i_private;
+
+  /* Return the address corresponding to the start of frame buffer. */
+
+  if (map->offset + map->length <= fb->fblen)
```

Comment:
after more thought, we need do more check:
```
if (map->offset >= 0 && map->offset < fb->fblen &&
   map->offset + map->length <= fb->fblen)
```
to avoid the hacker passes the wrong value. Other places need make the similar change.